amazon/ion/writer_binary.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at:
+#
+#    http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Binary Ion writer with symbol table management."""
+
+# Python 2/3 compatibility
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from .exceptions import IonException
+from .symbols import SID_ION_SYMBOL_TABLE, SID_IMPORTS, SHARED_TABLE_TYPE, \
+    SID_NAME, SID_VERSION, SID_MAX_ID, SID_SYMBOLS, SymbolTable, LOCAL_TABLE_TYPE
+from .writer_binary_raw import _raw_binary_writer, _WRITER_EVENT_NEEDS_INPUT_EMPTY,\
+    _WRITER_EVENT_COMPLETE_EMPTY
+from .writer_buffer import BufferTree
+from .core import IonEventType, IonType, IonEvent, DataEvent, Transition
+from .util import coroutine, Enum, record
+from .writer import writer_trampoline, _drain, partial_transition, WriteEventType
+
+
+_IVM = bytearray([0xE0, 0x01, 0x00, 0xEA])
```

Comment:
Does this need to be `bytearray`, why not an immutable `bytes` sequence as `b'\xE0\x01\x00\xEA'`